chore: refactored code

xnacly opened the pull request:
see commits

xnacly committed:
refactor:
- merged and removed duplicate code
- improved type security

xnacly committed:
fix: getPkgMgr not returning a package manager

- return in getPkgMgr after the loop overwrote the detected package manager to be none

xnacly committed:
doc: added comment
 -  indicate the order of detecting packagemanagers

Intevel commented on src/index.ts:
```diff
@@ -3,23 +3,38 @@ import consola from "consola";
+    return pkgMgr;
   } else {
-    consola.error("Can't detect any package manager");
+    consola.warn("Can't detect a package manager, auto fallback to: 'npm'");
```
`consola.error`

xnacly replied:
I specifically used warn.  A warning is smarter to indicate something went wrong, but due to the fallback the user can still be sure the package was installed. Throwing an error would suggest the installing failed. The fallback to npm is useful due to many VCS users ignoring the `package-lock.json` file, even when they are using npm as a pkg-manager.

Intevel replied:
Yes, but I wanted to make sure that the fallback is in the interest of the package user. So please add an option to activate or deactivate it.

xnacly replied:
will do

Intevel commented on src/index.ts:
```diff
@@ -3,23 +3,38 @@ import consola from "consola";
   } else {
-    consola.error("Can't detect any package manager");
+    consola.warn("Can't detect a package manager, auto fallback to: 'npm'");
+    return "npm";
```
Make an option to activate/deactivate fallback.

xnacly replied:
How should that be implemented? like the optional `cwd` flag?

Intevel replied:
Yes

xnacly committed:
feat: added fallbackToNpm option
- added the option to the docs
- added the option to installpackage()
- added the option to detectpackagemanager()

xnacly committed:
Delete package-lock.json

xnacly committed:
Update package.json

xnacly committed:
Delete test.js

xnacly committed:
Revert "Delete test.js"

This reverts commit 316f3b7eae052196ae8b4685f070506c6eb5f0e5.

xnacly committed:
Revert "Update package.json"

This reverts commit e1e0c2c96f92fbce26622bf7b4ecce4c0aed3c84.

xnacly committed:
Revert "Delete package-lock.json"

This reverts commit 0fed39e788c52c21ee35d8c7bf8f7ff321f78325.

xnacly committed:
Revert "feat: added fallbackToNpm option"

This reverts commit 4466378be06f71b632a4a75fff49b86f63c7104c.

xnacly committed:
feat: added fallbackToNpm option
    - added the option to the docs
    - added the option to installpackage()
    - added the option to detectpackagemanager()

Intevel reviewed (approved):
Thank you for this PR. Good work 💚
